content/blog/2019/06/2019-06-25-phase-1-multibranch-pipeline-support-for-gitlab.adoc

```diff
@@ -0,0 +1,305 @@
+
+. Select `$<root_dir>/target/gitlab-branch-source.hpi`
+
+. Select `Upload` 
+
+. Select `Install without restart`
+    
+== Usage
+
+Assuming plugin installation has done been already.
+
+=== Setting up GitLab Server Configuration on Jenkins
+
+. On jenkins, select `Manage Jenkins`
+
+. Select `Configure System`
+
+. Scroll down to find the `GitLab` section
+
+. Select `Add GitLab Server` | Select `GitLab Server`
+
+. Now you will now see the GitLab Server Configuration options. There are 4 fields that needs to be configured:
+    
+** `Name` - Plugin automatically generates an unique server name for you. User may want to configure this field
+to suit their needs but should make sure it is sufficiently unique. We recommend to keep it as it is.
+    
+** `Server URL` - Contains the URL to your GitLab Server. By default it is set to "https://gitlab.com". User can
+modify it to enter their GitLab Server URL e.g. https://gitlab.gnome.org/, http://gitlab.example.com:7990. etc.
+
+** `Credentials` - Contains a list of credentials entries that are of type GitLab Personal Access Token. When
+no credential has been added it shows "-none-". User can add a credential by clicking "Add" button.
+
+** `Web Hook` - This field is a checkbox. If you want the plugin to setup a webhook on your GitLab project(s)
+related jobs, check this box. The plugin listens to a URL for the concerned GitLab project(s) and when an event
+occurs in the GitLab Server, the server sends an event trigger to the URL where the web hook is setup. If you
+want continuous integration (or continuous delivery) on your GitLab project then you may want to automatically
+set it up.
+     
+. Adding a Personal Access Token Credentials (To automatically generate Personal Access Token see
+link:#creating-personal-access-token-within-jenkins[next section]):
+
+.. User is required to add a `GitLab Personal Access Token` type credentials entry to securely persist the token
+inside Jenkins.
+
+.. Generate a `Personal Access Token` on your GitLab Server:
+
+... Select profile dropdown menu from top-right corner
+
+... Select `Settings`
+
+... Select `Access Token` from left column
+
+... Enter a name | Set Scope to `api`,`read_user`, `read_repository`
+
+... Select `Create Personal Access Token`
+
+... Copy the token generated
+    
+.. Return to Jenkins | Select `Add` in Credentials field | Select `Jenkins`
+
+.. Set `Kind` to GitLab Personal Access Token
+
+.. Enter `Token`
+
+.. Enter a unique id in `ID`
+
+.. Enter a human readable description
+
+.. Select `Add`
+
+. Testing connection:
+
+.. Select your desired token in the `Credentials` dropdown
+
+.. Select `Test Connection`
+
+.. It should return something like `Credentials verified for user <username>`
+
+. Select `Apply` (at the bottom)
+
+. GitLab Server is now setup on Jenkins
+
+image::/images/post-images/2019-06-25-multibranch-pipeline-support-for-gitlab/gitlab-section.png[gitlab-section]
```

Comment:
I liked it better when these images were integrated into the corresponding steps. Would you like me to put them back where you had them?

Reply:
That would be great @bitwiseman.

Reply:
Here you go.
<img width="490" alt="Screen Shot 2019-06-26 at 10 42 23 AM" src="https://user-images.githubusercontent.com/1958953/60202488-837d6a80-97ff-11e9-9c5d-da50d6685425.png">

Reply:
Thanks a lot for helping me figure this out. :)